### DISCUSSION TITLE
Semantic highlighting?

### DISCUSSION THREAD

**Discussion by susliko** [scalameta/nvim-metals#611] [General]
In recent Metals releases semantic highlighting became enabled by default
In my setup it just highlights all the references with the same color

`enableSemanticHighlighting = false`: 
<img width="189" alt="image" src="https://github.com/scalameta/nvim-metals/assets/15569000/af5a5259-a25c-4e56-a714-235985a9f91c">

`enableSemanticHighlighting = true`: 
<img width="223" alt="image" src="https://github.com/scalameta/nvim-metals/assets/15569000/ae1ca49d-af26-454d-925d-285b3303c785">

Did anyone get it working properly? Maybe I'm missing some highlight group configuration or there is an issue with Metals itself?

**Comment by pepper-lees** [comment]
Do you have another highlighting plugin, like `nvim-treesitter` installed? I think that the colors you're seeing when `enableSemanticHighlighting = false` are from another syntax plugin, like Treesitter, while the colors you're seeing when `enableSemanticHighlighting = true` are from Metals. See #603
